### PR TITLE
[PM-4816] Create shared LoginApprovalComponent

### PR DESCRIPTION
**Pull request by alec-livefront**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-4816?atlOrigin=eyJpIjoiMzE0Yjg5YjFhYTViNGY0M2E4M2YyY2MyNjg3YTFhMWYiLCJwIjoiaiJ9

## 📔 Objective

Moves the `LoginApprovalComponent` to `libs/auth` and moves desktop-specific logic to a service.

## 📸 Screenshots

https://github.com/user-attachments/assets/4a6d98b8-fbfd-4507-8182-48ac12b68d9b

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: libs/auth/src/common/abstractions/login-approval.service.abstraction.ts
Comment (JaredSnider-Bitwarden) [inline libs/auth/src/common/abstractions/login-approval.service.abstraction.ts:1]: Can we rename this to be `LoginApprovalComponentServiceAbstraction`?

File: apps/desktop/src/auth/login/desktop-login-approval.service.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export class DesktopLoginApprovalService
+  extends DefaultLoginApprovalService
+  implements LoginApprovalServiceAbstraction
+{
+  constructor(private i18nService: I18nServiceAbstraction) {
+    super();
+  }
+
+  async onInit(loginApprovalComponent: LoginApprovalComponent): Promise<void> {
+    await ipc.auth.loginRequest(
```
Comment (JaredSnider-Bitwarden) [inline apps/desktop/src/auth/login/desktop-login-approval.service.ts:17]: This has lost the original `isVisible` logic. This should only fire if the desktop window is  not visible.

File: apps/desktop/src/auth/login/desktop-login-approval.service.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+@Injectable()
+export class DesktopLoginApprovalService
+  extends DefaultLoginApprovalService
+  implements LoginApprovalServiceAbstraction
+{
+  constructor(private i18nService: I18nServiceAbstraction) {
+    super();
+  }
+
+  async onInit(loginApprovalComponent: LoginApprovalComponent): Promise<void> {
```
Comment (JaredSnider-Bitwarden) [inline apps/desktop/src/auth/login/desktop-login-approval.service.ts:16]: I think we can avoid passing the component as a parameter here. Can we rename this method to be something like: `showLoginRequestedAlertIfWindowNotVisible()` and keep it as a no-op in the default implementation?

**Review by JaredSnider-Bitwarden — CHANGES_REQUESTED** [review]
Great start on this! Some feedback below:

File: libs/auth/src/common/abstractions/login-approval.service.abstraction.ts
Comment (alec-livefront) [reply libs/auth/src/common/abstractions/login-approval.service.abstraction.ts:1]: [a06d401](https://github.com/bitwarden/clients/pull/11982/commits/a06d401b6c4071cdfe1fec2baf66f24829e5f53a)

Should we update the other classes to contain `Component` too?

File: apps/desktop/src/auth/login/desktop-login-approval.service.ts (hunk 2 above)
Comment (JaredSnider-Bitwarden) [reply apps/desktop/src/auth/login/desktop-login-approval.service.ts:16]: Yes, passing an individual email vs the entire component context is a better solution.

File: apps/desktop/src/auth/login/desktop-login-approval.service.ts (hunk 1 above)
Comment (alec-livefront) [reply apps/desktop/src/auth/login/desktop-login-approval.service.ts]: 89ab519fe15f85eb795509f39643bea83cc2c1f9

File: libs/auth/src/common/abstractions/login-approval.service.abstraction.ts
Comment (JaredSnider-Bitwarden) [reply libs/auth/src/common/abstractions/login-approval.service.abstraction.ts:1]: Yes, please.

Comment (alec-livefront) [reply libs/auth/src/common/abstractions/login-approval.service.abstraction.ts:1]: [f0ccf69](https://github.com/bitwarden/clients/pull/11982/commits/f0ccf696647d5346900c7b77714dd1d0b3889ac8)

**Comment by micahblut** [comment]
@alec-livefront I notice this doesn't show the environment the request is coming from like the design does. It also looks like the spacing between "login attempt by" and "fingerprint phrase" is off. Finally, could you send me a screenshot with the updated styles?

**Review by JaredSnider-Bitwarden — APPROVED** [review]
LGTM! Nice testing work as well!

**Comment by alec-livefront** [comment]
@micahblut the original screenshot didn't have the correct spacing - I've updated with a screen recording which should show the correct spacing now. Could you elaborate on the environment? I'm not sure I see that in the screenshot attached to the ticket/

**Review by justindbaur — APPROVED** [review]

**Pull request merged**